### PR TITLE
Add category store

### PR DESCRIPTION
**Pull request by it-spiderman**

### CODE REVIEW

File: bootstrap.php [hunk 1]
```diff
@@ -4,7 +4,7 @@
 	return;
 }
 
-define( 'MWSTAKE_MEDIAWIKI_COMPONENT_COMMONWEBAPIS_VERSION', '2.0.14' );
+define( 'MWSTAKE_MEDIAWIKI_COMPONENT_COMMONWEBAPIS_VERSION', '2.0.15' );
```
Comment (osnard) [inline bootstrap.php:7]: Looks more like a minor-level change

File: src/Data/CategoryQueryStore/PrimaryDataProvider.php [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+	 */
+	protected function makePreFilterConds( ReaderParams $params ) {
+		if ( !$this->parentDone ) {
+			return parent::makePreFilterConds( $params );
+		}
+		$query = $params->getQuery();
+		$conds = [];
+		if ( $query ) {
+			$query = mb_strtolower( str_replace( ' ', '_', $query ) );
+			$field = "REPLACE( LOWER( CONVERT(cl_to USING utf8mb4) ), '_', ' ' )";
```
Comment (osnard) [inline src/Data/CategoryQueryStore/PrimaryDataProvider.php:65]: Would there be a way of doing this DBMS agnostic?

**Review by osnard — CHANGES_REQUESTED** [review]

File: src/Data/CategoryQueryStore/PrimaryDataProvider.php (hunk 2 above)
Comment (it-spiderman) [reply src/Data/CategoryQueryStore/PrimaryDataProvider.php]: using index now

File: bootstrap.php (hunk 1 above)
Comment (it-spiderman) [reply bootstrap.php:7]: why? it just adds more of  the same, not changing anything

**Pull request merged**